### PR TITLE
Share results modal for benefit eligibility questionnaire

### PR DESCRIPTION
**Pull request by vetskrieg**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

Share results modal is added to the benefit eligibility questionnaire.

## Related issue(s)

[PTEMSVT-87](https://jira.devops.va.gov/browse/PTEMSVT-87)

## Testing done

Local testing.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  ||image|
| Desktop ||image|

## What areas of the site does it impact?

Benefit eligibility questionnaire which is currently disabled.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <VaModal
+        id="share-results-modal"
+        onCloseEvent={() => {
+          setShowAlert(false);
+          toggleModal(false);
+        }}
+        modalTitle="Save, bookmark, or share your results"
+        initialFocusSelector="#va-modal-title"
+        visible={isOpen}
+        uswds
```
Comment (jamigibbs) [inline src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx:35]: The `uswds` prop is no longer necessary for design system web components. They will all default to using the uswds v3 variation now.

File: src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        large
+      >
+        <p>Copy link</p>
+        <input
```
Comment (jamigibbs) [inline src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx:39]: Can the `va-text-input` component be used here? https://design.va.gov/storybook/?path=/docs/uswds-va-text-input--docs

**Review by jamigibbs — DISMISSED** [review]

File: src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx (hunk 2 above)
Comment (vetskrieg) [reply src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx]: Yes, this has been updated.

File: src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx (hunk 1 above)
Comment (vetskrieg) [reply src/applications/benefit-eligibility-questionnaire/components/ShareResultsModal.jsx]: Ok

**Review by ghost — APPROVED** [review]
LGTM

**Review by Andrew565 — APPROVED** [review]

**Pull request merged**